### ISSUE TITLE
DeprecationWarning for datetime methods in Python 3.12

### ISSUE THREAD

**Issue by dsotirho-ucsc** [DataBiosphere/azul#5953]
Upgrading to Python 3.12 (https://github.com/DataBiosphere/azul/issues/5736) introduces DeprecationWarning with the `datetime` methods `utcnow` and `utcfromtimestamp`.

https://docs.python.org/3/library/datetime.html#datetime.datetime.utcnow

> Warning Because naive datetime objects are treated by many datetime methods as local times, it is preferred to use aware datetimes to represent times in UTC. As such, the recommended way to create an object representing the current time in UTC is by calling datetime.now(timezone.utc).
> Deprecated since version 3.12: Use datetime.now() with UTC instead.

https://docs.python.org/3/library/datetime.html#datetime.datetime.utcfromtimestamp

> Warning Because naive datetime objects are treated by many datetime methods as local times, it is preferred to use aware datetimes to represent times in UTC. As such, the recommended way to create an object representing a specific timestamp in UTC is by calling datetime.fromtimestamp(timestamp, tz=timezone.utc).
> Deprecated since version 3.12: Use datetime.fromtimestamp() with UTC instead.

`datetime.utcnow` was found in Azul's codebase and will need to be updated.

In addition, the following imported dependencies were also found to make use of the deprecated `datetime` methods:
- .venv/lib/python3.12/site-packages/botocore/auth.py
- .venv/lib/python3.12/site-packages/google/api_core/datetime_helpers.py
- .venv/lib/python3.12/site-packages/moto/core/utils.py
- .venv/lib/python3.12/site-packages/moto/s3/models.py
- .venv/lib/python3.12/site-packages/moto/dynamodb/models/table.py

Blocked by:
- https://github.com/boto/boto3/issues/3889

**Comment by dsotirho-ucsc** [comment]
@hannes-ucsc: "Update `moto` and `google-api-core` as part of #5736 if that is easy to do. Link to blocking boto/botocore ticket in description. We will park this until those are resolved at which point we will also update our own code. According to the botocore team, the removal of `utcnow()` et al is slated for Python 3.16 so we have some time."

**Comment by achave11-ucsc** [comment]
This is parked because blocker is parked.

**Comment by dsotirho-ucsc** [comment]
Moving to Triage since blocker #5736 has been merged.

**Comment by dsotirho-ucsc** [comment]
Wait for upstream blocker in Boto to be resolved.

**Comment by dsotirho-ucsc** [comment]
Upstream Boto blocker has been closed as completed, however I do not see a PR or commit that addresses the issue.

Update: Resolving PR https://github.com/boto/botocore/pull/3239 has been identified.

**Comment by dsotirho-ucsc** [comment]
Assignee to perform the following tasks:
- [x] Update Azul to use the proposed alternatives to the deprecated functionality
- [x] Confirm that Azul depends on a version of boto/botocore that includes the fix
    - Fixed in boto3 1.40.2, we are currently on 1.42.69
- [x] Survey Azul's dependencies for usages of the deprecated functionality

## Usage of deprecated methods in current dependencies

| package | version | depcrecated method | file |
| -- | -- | -- | -- |
| aws_requests_auth | 0.4.3 | datetime.utcnow() | aws_requests_auth/aws_auth.py |
| chalice | 1.32.0+20 | datetime.utcfromtimestamp() | chalice/app.py |
| chalice | 1.32.0+20 | datetime.utcfromtimestamp() | chalice/awsclient.py |
| chalice | 1.32.0+20 | datetime.utcnow() | chalice/logs.py |
| chalice | 1.32.0+20 | datetime.utcnow() | chalice/utils.py |
| opensearch-py | 2.8.0 | datetime.utcfromtimestamp() | opensearchpy/helpers/faceted_search.py |
| opensearch-py | 2.8.0 | datetime.utcfromtimestamp() | opensearchpy/helpers/field.py |
| requests_toolbelt | 1.0.0 | datetime.utcnow() | requests_toolbelt/adapters/x509.py |

**Comment by dsotirho-ucsc** [comment]
Regarding the [four Azul dependencies](https://github.com/DataBiosphere/azul/issues/5953#issuecomment-3202050379) that use the deprecated datetime methods:


- **aws_requests_auth** : We only use this in /src/azul/opensearch.py and should investigate if we can remove it as a  dependency
- **chalice** : We can update our fork of Chalice to use the non-deprecated functionality.
- **opensearch-py** : The deprecated methods are located in the `faceted_search` helper, and the Date & DateRange helpers. We don't use any of these.
- **requests_toolbelt** : The deprecated methods are located in X.509 functionality which we don't use.

**Comment by dsotirho-ucsc** [comment]
Partial PR https://github.com/DataBiosphere/azul/pull/7904 was merged.

**Comment by dsotirho-ucsc** [comment]
Assignee to implement solution for [remaining issue ](https://github.com/DataBiosphere/azul/issues/5953#issuecomment-4195501493)with dependencies

**Comment by hannes-ucsc** [comment]
Balancing work load between @dsotirho-ucsc and @achave11-ucsc

**Comment by hannes-ucsc** [comment]
For demo, show in one main deployment that neither the IT logs nor application logs contain a related deprecation warning.
